### PR TITLE
build: automate release using GH release and circle ci

### PR DESCRIPTION
**Pull request by dariuszkuc**
Trigger CircleCI release job when new GH release (and associated tag) are created

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -49,6 +58,23 @@ workflows:
...
+  release:
+    jobs:
+      - test:
+          name: Test (<< matrix.rust_channel >> rust on << matrix.platform >>)
+          matrix:
+            parameters:
+              platform: [linux]
+              rust_channel: [stable]
+          <<: *crate_release
+      - publish:
```
Comment (sachindshinde) [inline .circleci/config.yml:70]: We'll want a `requires` key on the `publish` job to depend on the `test` jobs, [similar to federation-rs](https://github.com/apollographql/federation-rs/blob/0757a29c9fb58b2c7f1c9e5ffd9f4eb2a88c6acd/.circleci/config.yml#L101).

File: .circleci/config.yml [hunk 2]
```diff
@@ -102,6 +126,24 @@ jobs:
...
+        default: stable
+      platform:
+        type: executor
+    executor: << parameters.platform >>
+    steps:
+      - checkout
+      - install_system_deps:
+          rust_channel: << parameters.rust_channel >>
+          platform: << parameters.platform >>
+      - run:
```
Comment (sachindshinde) [inline .circleci/config.yml:143]: Presumably we'd need to run `cargo login -- $CRATES_IO_API_TOKEN` before doing publish, while putting `CRATES_IO_API_TOKEN` into some new CircleCI context (or if there's already some CircleCI context that the `apollographql` org uses for publishing, we could use that). Or if Rust folks are doing something else to auth with `crates.io`, we can use that.

**Review by sachindshinde — CHANGES_REQUESTED** [review]

File: .circleci/config.yml (hunk 2 above)
Comment (dariuszkuc) [reply .circleci/config.yml:143]: I was wondering about this -> I based this on federation-rs where I dont see any explicit login setup. I'm guessing it is already configured on circle ci.

File: .circleci/config.yml (hunk 1 above)
Comment (dariuszkuc) [reply .circleci/config.yml:70]: was wondering whether we should just collapse it to a single thing (so we wouldn't have to setup env twice). might do this later

**Pull request merged**